File: src/main/java/com/awsblog/queueing/test/PeekSDK.java
package com.awsblog.queueing.test;

import com.awsblog.queueing.sdk.Dynamodb;
import com.awsblog.queueing.sdk.IDbPriorityQueue;
import com.awsblog.queueing.model.PeekResult;
import com.awsblog.queueing.model.QueueStats;
import com.awsblog.queueing.sdk.QueueSdkClient;
import com.awsblog.queueing.utils.Utils;

/**
 * Simple test app for peeking the record from the queue
 * 
 * @author zorani
 *
 */
public class PeekSDK {

	/**
	 * @param args
	 */
	public static void main(String[] args) {

		IDbPriorityQueue<Dynamodb> client = new Dynamodb.Builder().withCredentialsProfileName("default")
				.withRegion("us-east-2")
				.withLogicalTableName("assignment_schedule")
				.build();
		
		QueueStats queueStats = client.getQueueStats();
		System.out.println(Utils.toJSON(queueStats));

		PeekResult peek = client.peek();
		System.out.println(Utils.toJSON(peek.getPeekedAssignmentId()));
		
		queueStats = client.getQueueStats();
		System.out.println(Utils.toJSON(queueStats));		
	}

} // end TestSDK

File: src/main/java/com/awsblog/queueing/test/PeekSDK.kt
package com.awsblog.queueing.test

import com.awsblog.queueing.sdk.QueueSdkClient
import com.awsblog.queueing.utils.Utils

/**
 * Simple test app for peeking the record from the queue
 *
 * @author zorani
 */
object PeekSDK {
    /**
     * @param args
     */
    @JvmStatic
    fun main(args: Array<String>) {
        val client = QueueSdkClient.Builder().withCredentialsProfileName("default")
            .withRegion("us-east-2")
            .withLogicalTableName("assignment_schedule")
            .build()
        var queueStats = client!!.queueStats
        println(Utils.toJSON(queueStats))
        val peek = client.peek()
        println(Utils.toJSON(peek.peekedAssignmentId))
        queueStats = client.queueStats
        println(Utils.toJSON(queueStats))
    }
} // end TestSDK
